addons/cloud_storage_azure/models/res_config_settings.py

```diff
@@ -0,0 +1,76 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import requests
+from datetime import datetime, timedelta, timezone
+
+from odoo import models, fields, _
+from odoo.exceptions import ValidationError
+
+from .cloud_storage_azure import CloudStorageAzure
+
+
+class CloudStorageSettings(models.TransientModel):
+    """
+    Instructions:
+    cloud_storage_azure_account_name, cloud_storage_azure_container_name:
+        if changed and old container names are still in use, you should
+        promise the current application registration has the permission
+        to access all old containers.
+    """
+    _inherit = 'res.config.settings'
+
+    cloud_storage_provider = fields.Selection(selection_add=[('azure', 'Azure Cloud Storage')])
+
+    cloud_storage_azure_account_name = fields.Char(
+        string='Azure Account Name',
+        config_parameter='cloud_storage_azure_account_name')
+    cloud_storage_azure_container_name = fields.Char(
+        string='Azure Container Name',
+        config_parameter='cloud_storage_azure_container_name')
+    # Application Registry Info
+    cloud_storage_azure_tenant_id = fields.Char(
+        string='Azure Tenant ID',
+        config_parameter='cloud_storage_azure_tenant_id')
+    cloud_storage_azure_client_id = fields.Char(
+        string='Azure Client ID',
+        config_parameter='cloud_storage_azure_client_id')
+    cloud_storage_azure_client_secret = fields.Char(
+        string='Azure Client Secret',
+        config_parameter='cloud_storage_azure_client_secret')
+
+    def _get_cloud_storage_configuration(self):
```

Comment:
This method is called everywhere with `sudo()`, and sometimes in very indirect ways.  Since the `sudo()` looks like some artificial necessity (for accessing config parameters), I suggest that it does the `sudo()` itself, instead of relying on the caller.  This will also make the calling sites less cumbersome.